File: core/src/com/nopalsoft/thetruecolor/handlers/GameServicesHandler.java
package com.nopalsoft.thetruecolor.handlers;

public interface GameServicesHandler {

    /**
     * This method abstracts away GPGS or AGC.
     */
    void submitScore(long score);

    /**
     * This method abstracts away GPGS or AGC.
     */
    void unlockAchievement(String achievementId);

    void unlockStepAchievement(float steps, String achievementID);

    /**
     * This method abstracts away GPGS or AGC.
     */
    void getLeaderboard();

    void getScores();

    /**
     * This method abstracts away GPGS or AGC.
     */
    void getAchievements();

    boolean isSignedIn();

    void signIn();

    void signOut();

}

File: core/src/com/nopalsoft/thetruecolor/handlers/GameServicesHandler.kt
package com.nopalsoft.thetruecolor.handlers

interface GameServicesHandler {

    // This method abstracts away GPGS or AGC.
    fun submitScore(score: Long)

    // This method abstracts away GPGS or AGC.
    fun unlockAchievement(achievementId: String?)

    fun unlockStepAchievement(steps: Float, achievementID: String?)

    // This method abstracts away GPGS or AGC.
    fun getLeaderboard()

    fun getScores()

    // This method abstracts away GPGS or AGC.
    fun getAchievements()

    fun isSignedIn(): Boolean

    fun signIn()

    fun signOut()
}